### PR TITLE
added 3.3+ compatibility

### PR DESCRIPTION
**Pull request by kybrdbnd**
change to line 14  in module test_fiximport.py

### CODE REVIEW

File: common/test_fiximports.py [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
+#returning functions
+def d_setUp():
+    return None
 
 class TestFixImports(unittest.TestCase):
 
     def setUp(self):
         self.fiximports = FixImports()
-        self.fiximports.printErrorMsg = lambda *_: None
+        ret_msg = d_setUp()
```
Comment (djmitche) [inline common/test_fiximports.py]: so `ret_msg` is None..

File: common/test_fiximports.py [hunk 2]
```diff
@@ -2,12 +2,16 @@
...
+        self.fiximports.printErrorMsg = ret_msg
```
Comment (djmitche) [inline common/test_fiximports.py]: which is not what `printErrorMessage` should be

File: slave/buildslave/commands/base.py [hunk 3]
```diff
@@ -147,7 +155,7 @@ def __init__(self, builder, stepId, args):
         self.args = args
         self.startTime = None
 
-        missingArgs = filter(lambda arg: arg not in args, self.requiredArgs)
```
Comment (djmitche) [inline slave/buildslave/commands/base.py]: How is this not Python-3.3 compatible?  Or PEP8?

**Comment by djmitche** [comment]
You seem to have mixed up PEP8 and Python-3.3 compatibility here.  As far as I know, `lambda` is still supported in Python-3.3.  PEP8 prohibits assigning lambdas directly to a variable, so the first hunk is a reasonable PEP8 violation to fix.

File: common/test_fiximports.py (hunk 1 above)
Comment (kybrdbnd) [inline common/test_fiximports.py]: Yes

File: slave/buildslave/commands/base.py (hunk 3 above)
Comment (kybrdbnd) [inline slave/buildslave/commands/base.py]: this is also 3.3+ compatible but not according to PEP8 standards. That's why i have replaced the lambdas with a def function.

Comment (djmitche) [inline slave/buildslave/commands/base.py]: What PEP8 standard does it violate?

**Comment by tardyp** [comment]
Is this *all* what we need for python 3.3 is buildbot slave module?

If yes, please add a .travisrc change so that python 3.3 is tested, and run the buildslave unit tests

**Comment by kybrdbnd** [comment]
my commit 'Python3.3 compatibility' means that the changes i have made are according to PEP8. That is i have replaced the lamdas with the def functions

File: common/test_fiximports.py (hunk 1 above)
Comment (djmitche) [inline common/test_fiximports.py]: And that's incorrect..

Comment (djmitche) [inline common/test_fiximports.py]: Note that the existing code assigned a function of any number of positional arguments that always returns None.  That's very different from the simple value None.

**Comment by djmitche** [comment]
That is a confusing name for a commit!  On that topic, there's also no need to indicate line numbers in the commit message -- the diff itself makes those clear.

**Comment by kybrdbnd** [comment]
@tardyp i have just checked some of the modules because in the mail that i have received you mentioned that you have to look some of the modules and fix the lambas, pull the requests so that we can evaluate it. That's all i have done it now

File: slave/buildslave/commands/base.py (hunk 3 above)
Comment (kybrdbnd) [inline slave/buildslave/commands/base.py]: the use of lambas ofcourse

File: common/test_fiximports.py (hunk 1 above)
Comment (kybrdbnd) [inline common/test_fiximports.py]: for that i can change def setUp(**krgs): return None

File: slave/buildslave/commands/base.py (hunk 3 above)
Comment (djmitche) [inline slave/buildslave/commands/base.py]: PEP8 doesn't forbid lambdas!  It forbids the assignment of a lambda directly to a local variable - the fifth bullet point in https://www.python.org/dev/peps/pep-0008/#programming-recommendations

So actually, neither of the lambdas you're addressing here violate that recommendation -- in `test_fiximports.py` the lambda is assigned to an instance attribute (not a bare identifier), while in `base.py` the lambda is not assigned to anything.

File: common/test_fiximports.py (hunk 1 above)
Comment (djmitche) [inline common/test_fiximports.py]: It sounds like that would be more correct; however, I don't think that PEP8 actually requires this change.

**Comment by djmitche** [comment]
This has gotten a little involved -- sorry about that!

I'd suggest that we close this pull request, and you begin a new one described as fixing PEP8-violating lambda expressions rather than Python-3.3 compatibility.  Here are a few examples you could fix (but start by addressing just one):
```
dustin@euclid ~/code/buildbot/t/buildbot [master] $ git  grep -E ' [a-z0-9_]+ = lambda'
master/buildbot/locks.py:    debuglog = lambda m: None
master/buildbot/process/buildstep.py:    _run_finished_hook = lambda self: None  # for tests
master/buildbot/process/buildstep.py:                callback = lambda cmd_arg, local_logname=logname: self.addLog(
master/buildbot/process/buildstep.py:                callback = lambda cmd_arg, logname=logname: self.addLog(
master/buildbot/status/status_gerrit.py:        callback = lambda gerrit_version: self.processVersion(gerrit_version, func)
master/buildbot/steps/source/svn.py:        quote = lambda uri: urllib.quote(uri, "!$&'()*+,-./:=@_~")
master/buildbot/steps/transfer.py:    callback = lambda arg: step.step_status.addLog('stdio')
master/buildbot/test/unit/test_config.py:        func = lambda _: "dummy"
master/buildbot/test/unit/test_config.py:        callable = lambda: None
master/buildbot/test/unit/test_config.py:        callable = lambda: None
master/buildbot/test/unit/test_config.py:        callable = lambda: None
master/buildbot/test/unit/test_config.py:        ns = lambda: 'ns'
master/buildbot/test/unit/test_config.py:        nb = lambda: 'nb'
master/buildbot/test/unit/test_process_builder.py:        cble = lambda: None
master/buildbot/test/unit/test_steps_trigger.py:        make_fake_br = lambda brid, builderid: fakedb.BuildRequest(
master/buildbot/test/unit/test_steps_trigger.py:        make_fake_build = lambda brid: fakedb.Build(
master/docs/manual/cfg-buildslaves-openstack.rst:        cmp_fn = lambda x,y: cmp(x.created, y.created)
```

**Comment by kybrdbnd** [comment]
okey

**Pull request closed without merging**